ingestion/functions/parsing/germany/germany.py

```diff
@@ -0,0 +1,111 @@
+import os
+import sys
+from datetime import date, datetime
+import csv
+
+# Layer code, like parsing_lib, is added to the path by AWS.
+# To test locally (e.g. via pytest), we have to modify sys.path.
+# pylint: disable=import-error
+if "lambda" not in sys.argv[0]:
+    sys.path.append(
+        os.path.join(
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__))), "common/python"
+        )
+    )
+import parsing_lib
+
+_DATE_INDEX = 8
+_AGE_INDEX = 4
+_GENDER_INDEX = 5
+_CASECOUNT = 6
+_DEATHCOUNT = 7
+_RECOVERYCOUNT = 15
+_ADMIN1 = 2
+_ADMIN2 = 3
+
+
+def convert_date(raw_date):
+    """
+    Convert raw date field into a value interpretable by the dataserver.
+
+    """
+    date = datetime.strptime(raw_date, "%Y/%m/%d %H:%M:%S")
+    return date.strftime("%m/%d/%YZ")
+
+
+def convert_gender(raw_gender):
+    if raw_gender == "M":
+        return "Male"
+    if raw_gender == "W":
+        return "Female"
+    return None
+
+
+def convert_demographics(gender: str, age: str):
+    demo = {}
+    if gender:
+        demo["gender"] = convert_gender(gender)
+    if age:
+        age_range = age.split("-")
+        if len(age_range) > 1:
+            demo["ageRange"] = {
+                "start": float("".join([x for x in age_range[0] if x.isdigit()])),
+                "end": float("".join([x for x in age_range[1] if x.isdigit()])),
+            }
+        else:
+            demo["ageRange"] = {
+                "start": float("".join([x for x in age_range[0] if x.isdigit()])),
+                "end": float("".join([x for x in age_range[0] if x.isdigit()])),
+            }
+    return demo
+
+
+def parse_cases(raw_data_file: str, source_id: str, source_url: str):
+    """
+    Parses G.h-format case data from raw API data.
+    """
+    with open(raw_data_file, "r") as f:
+        reader = csv.reader(f)
+        next(reader)  # Skip the header.
+        cases = []
+        for row in reader:
+            num_confirmed_cases = int(row[_CASECOUNT])
+            if num_confirmed_cases < 1:
+                continue
+            try:
+                case = {
+                    "caseReference": {"sourceId": source_id, "sourceUrl": source_url},
+                    "location": {
+                        "country": "Germany",
```

Comment:
I will clarify this in the docs for writing parsers today but parsers should only output a query, they do not need to pass anything else, in your case you know that's always admin2 so you can also pass `limitToResolution: 'Admin2'` in the location dictionary which will restrict geocodes to admin2.

Reply:
thanks! do you know which of the parser docs are the most recent? I've been working off of the Swagger API docs (great for most things but no expected values) and the curator bulk upload docs https://github.com/globaldothealth/list/tree/main/verification/curator-service/ui (good expected values but incomplete set of fields)

Reply:
if you run a stack locally you can go to localhost:3001/api-docs but I've just reloaded dev with the latest changes from this morning, this should be quite comprehensive, let us know if there's something missing from the API docs there.

https://dev-curator.ghdsi.org/api-docs/

Reply:
this is great, thanks a ton